src/main/java/com/codecool/expertsystem/models/RuleParser.java

```diff
@@ -17,6 +17,8 @@ public RuleParser(String xmlPath) {
     public RuleRepository getRuleRepository() {
         RuleRepository ruleRepository = new RuleRepository();
         NodeList nodeList = doc.getElementsByTagName("Rule");
```

Comment:
Have you initialized doc somewhere else?

Reply:
doc is initialized in XMLParser in loadXMLDocument method